src/LittleBitsR2Controller/Services/BluetoothService.cs

```diff
@@ -0,0 +1,185 @@
+using Plugin.BLE;
+using Plugin.BLE.Abstractions.Contracts;
+using Plugin.BLE.Abstractions.EventArgs;
+
+namespace LittleBitsR2Controller.Services;
+
+public class BluetoothService : IBluetoothService, IDisposable
+{
+    private readonly IBluetoothLE _bluetoothLE;
+    private readonly IAdapter _adapter;
+    private IDevice? _connectedDevice;
+    private ICharacteristic? _writeCharacteristic;
+    private bool _disposed;
+    
+    // R2D2 Service and Characteristic UUIDs based on reverse engineering
+    // These UUIDs are typical for BLE devices, may need adjustment based on actual device
+    private const string ServiceUuid = "0000ffe0-0000-1000-8000-00805f9b34fb";
+    private const string CharacteristicUuid = "0000ffe1-0000-1000-8000-00805f9b34fb";
+    
+    public event EventHandler<bool>? ConnectionStatusChanged;
```

Comment:
This could have a memory leak. Follow the guidance on https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/messagingcenter?view=net-maui-10.0 to avoid it.